### PR TITLE
PWX-27646: Block cloudsnaps on FA direct access volumes

### PR DESCRIPTION
**Pull request by ashinde-px**
**What this PR does / why we need it**:
Block cloudsnaps on FA direct access volumes

**Which issue(s) this PR fixes** (optional)
Closes #PWX-27646

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ashinde-px** [comment]
Updated testing results in the ticket comment : https://portworx.atlassian.net/browse/PWX-27646

**Review by rkinhalkar-px — APPROVED** [review]

File: csi/controller.go [hunk 1]
```diff
@@ -1032,11 +1032,15 @@ func (s *OsdCsiServer) createCloudBackup(
...
 
 	// Get any labels passed in by the CO
-	_, locator, _, err := s.specHandler.SpecFromOpts(req.GetParameters())
+	spec, locator, _, err := s.specHandler.SpecFromOpts(req.GetParameters())
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "Unable to get parameters: %v", err)
 	}
 
+	if spec.IsPureVolume() {
+		return nil, status.Errorf(codes.InvalidArgument, "cloudsnap feature is not supported for pure volumes")
```
Comment (ggriffiths) [inline csi/controller.go:1041]: nit:
```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "cloudsnaps are not supported for pure volumes")
```

Comment (ashinde-px) [reply csi/controller.go]: Done.

**Review by adityadani — CHANGES_REQUESTED** [review]
@ashinde-px This specific change for taking cloud backups through CSI snapshots has not been released yet. To block cloudsnaps on FA volumes, you need to make a change in porx so that you block cloudsnap creation triggered through CLI (and not CSI)

**Comment by ashinde-px** [comment]
> @ashinde-px This specific change for taking cloud backups through CSI snapshots has not been released yet. To block cloudsnaps on FA volumes, you need to make a change in porx so that you block cloudsnap creation triggered through CLI (and not CSI)

Hi @adityadani sure let me create PR at porx side then to block request from CLI, was waited to merge this PR so that to add both changes at once. Closing this PR as its not required.

**Comment by ashinde-px** [comment]
let me create PR at porx side then to block request from CLI, was waited to merge this PR so that to add both changes at once. Closing this PR as its not required.

**Pull request closed without merging**